### PR TITLE
Fix createChannelBuilders not using overriden label/description

### PR DESCRIPTION
**Pull request by arjanmels**
Fixes createChannelBuilders not using the overridden label/description (#2225)

Below example would lead to a channel with label "Original Label", with the new code it will lead to "Label Overridden" as expected.

thing-types.xml
```
	<channel-group-type id="channel_group_type_example">
		<label>Channel Group Type Example</label>
		<channels>
			<channel id="channel_example" typeId="channel_type_example">
				<label>Label Overridden</label>
			</channel>
                </channels>
        </channel-group-type>

        <channel-type id="channel_type_example">
		<item-type>String</item-type>
		<label>Original Label</label>
	</channel-type>
```

In the ThingHandler:
```
    void updateChannels() {
        List<Channel> toBeAddedChannels = new ArrayList<>();
        toBeAddedChannels.addAll(createChannelsForGroup(groupID, CHANNEL_GROUP_TYPE_EXTRUDER));
    }

    private List<Channel> createChannelsForGroup(String channelGroupId, ChannelGroupTypeUID channelGroupTypeUID) {
        List<Channel> channels = new ArrayList<>();
        ThingHandlerCallback callback = getCallback();
        if (callback != null) {
            for (ChannelBuilder channelBuilder : callback.createChannelBuilders(
                    new ChannelGroupUID(getThing().getUID(), channelGroupId), channelGroupTypeUID)) {
                channels.add(channelBuilder.build());
            }
        }
        return channels;
    }
```

### CODE REVIEW

**Comment by cweitkamp** [comment]
Thanks for this piece of code. Can you add an example in the description of the issue and a unit test here?

**Comment by arjanmels** [comment]
Added an example to the description. Is it ok like this?
On the unit test: not certain where and how to do this: I think it would require an xml example. I am afraid this might be beyond my comprehension of this project.

**Comment by cweitkamp** [comment]
I think the best place to add new unit test is the [ThingManagerOSGiJavaTest](https://github.com/openhab/openhab-core/blob/127724c0e31ad8abf0855d87fbf96204267a39be/itests/org.openhab.core.thing.tests/src/main/java/org/openhab/core/thing/internal/ThingManagerOSGiJavaTest.java). wdyt?

**Comment by arjanmels** [comment]
Thanks for the hint: yes that looks good. Test implemented.

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java [hunk 1]
```diff
@@ -336,8 +336,10 @@ public ChannelBuilder editChannel(Thing thing, ChannelUID channelUID) {
                 ChannelType channelType = channelTypeRegistry.getChannelType(channelDefinition.getChannelTypeUID());
                 if (channelType != null) {
                     ChannelUID channelUID = new ChannelUID(channelGroupUID, channelDefinition.getId());
-                    channelBuilders.add(ThingFactoryHelper.createChannelBuilder(channelUID, channelType,
-                            configDescriptionRegistry));
+                    ChannelBuilder channelBuilder = ThingFactoryHelper.createChannelBuilder(channelUID,
+                            channelDefinition, configDescriptionRegistry);
+                    if (channelBuilder != null)
+                        channelBuilders.add(channelBuilder);
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java:342]:
```suggestion
                    if (channelBuilder != null) {
                        channelBuilders.add(channelBuilder);
                    }
```

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingFactoryHelper.java [hunk 2]
```diff
@@ -134,46 +134,68 @@
...
+        });
+        if (channelType == null) {
+            logger.warn("Could not create channel '{}', because channel type '{}' could not be found.",
+                    channelDefinition.getId(), channelDefinition.getChannelTypeUID());
+            return null;
+        }
+
+        String label = channelDefinition.getLabel();
+        if (label == null)
+            label = channelType.getLabel();
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingFactoryHelper.java:186]:
```suggestion
        if (label == null) {
            label = channelType.getLabel();
        }
```

**Review by cweitkamp — COMMENTED** [review]
Thanks for adding the unit test. It shows that your code is working. Nice.

Two minor comments on formatting. Please add brackets in one line `if`-statements,

**Comment by arjanmels** [comment]
Great, thanks. Updates made.

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**